## Test results — PR #276 (latest: ac934e4)

- [x] `go build ./...` — clean
- [x] `go vet ./...` — no issues
- [ ] `go test ./...` — **1 failure in `internal/cmd`**

```
ok  github.com/arun-gupta/agentctl/cmd/agentctl          0.325s
ok  github.com/arun-gupta/agentctl/internal/adapters     0.517s
FAIL github.com/arun-gupta/agentctl/internal/cmd        36.753s
ok  github.com/arun-gupta/agentctl/internal/config       0.552s
ok  github.com/arun-gupta/agentctl/internal/diagnostics  0.939s
ok  github.com/arun-gupta/agentctl/internal/git          2.062s
ok  github.com/arun-gupta/agentctl/internal/notify
ok  github.com/arun-gupta/agentctl/internal/process
ok  github.com/arun-gupta/agentctl/internal/sdd          1.588s
ok  github.com/arun-gupta/agentctl/internal/state
ok  github.com/arun-gupta/agentctl/internal/vcs          1.630s
?   github.com/arun-gupta/agentctl/internal/xdg          [no test files]
```

### Failing test: `TestRunInfo_ghVersionErrorShown`

```
--- FAIL: TestRunInfo_ghVersionErrorShown (0.00s)
    info_test.go:319: did not expect success marker when gh --version fails; got:
        ...
        Coding Agents:
          ✓ claude       (default)
          ✓ codex
          ...
```

**Root cause:** The test asserts `strings.Contains(out, "✓")` is false when `gh --version` fails. But `lookPathFn` (the stubbed LookPath) is only used for the `gh` check in `info.go`; `adapter.CheckBinary()` calls `exec.LookPath` directly and is not stubbed. So all actually-installed agents still emit `✓` in the Coding Agents section, causing the broad `✓`-anywhere assertion to fire.

**Fix:** Narrow the assertion to check only the `GitHub CLI:` line, not the entire output:

```go
for _, line := range strings.Split(out, "\n") {
    if strings.Contains(line, "GitHub CLI") &&
        (strings.Contains(line, "✓") || strings.Contains(line, "OK")) {
        t.Errorf("did not expect success marker in GitHub CLI line; got: %s", line)
    }
}
```

---
*Reviewed by Claude Sonnet 4.6*

--- a/internal/cmd/info_test.go
+++ b/internal/cmd/info_test.go
@@ -315,8 +315,10 @@ out := buf.String()
 if !strings.Contains(out, "GitHub CLI: error running gh --version:") {
 t.Errorf("expected gh version error indicator; got:\n%s", out)
 }
-if strings.Contains(out, "✓") || strings.Contains(out, "OK") {
-t.Errorf("did not expect success marker when gh --version fails; got:\n%s", out)
+for _, line := range strings.Split(out, "\n") {
+	if strings.Contains(line, "GitHub CLI") && (strings.Contains(line, "✓") || strings.Contains(line, "OK")) {
+		t.Errorf("did not expect success marker in GitHub CLI line; got: %s", line)
+	}
 }
 }
 

